src/binding.cc

```diff
@@ -357,16 +389,45 @@ void deno_set_flags(int* argc, char** argv) {
 
 const char* deno_last_exception(Deno* d) { return d->last_exception.c_str(); }
 
-int deno_execute(Deno* d, const char* js_filename, const char* js_source) {
+static int execute_js(Deno* d, const char* js_filename, const char* js_source) {
   auto* isolate = d->isolate;
   v8::Locker locker(isolate);
   v8::Isolate::Scope isolate_scope(isolate);
   v8::HandleScope handle_scope(isolate);
   auto context = d->context.Get(d->isolate);
   return deno::Execute(context, js_filename, js_source) ? 1 : 0;
+  // TODO: process incoming async messages.
+}
+
+static void backend_main(Deno* d) {
+  deno_buf msg = DENO_BUF_INIT;
+  while (d->cmd_queue.Recv(&msg)) {
+    d->recv_cb(d, &msg);
+    // If the callback needs the keep the buffer around after the callback
+    // returns, it can take owenership of the buffer with `deno_buf_move()`.
+    deno_buf_delete(&msg);
+  }
+}
+
+int deno_execute(Deno* d, const char* js_filename, const char* js_source) {
+  if (!d->threads_enabled) {
+    return execute_js(d, js_filename, js_source);
+  }
+
+  // Start backend worker thread.
+  // TODO: use multiple backend threads.
+  std::thread backend_thread(backend_main, d);
+  auto r = execute_js(d, js_filename, js_source);
+  // TODO: join the backend thread.
```

Comment:
The backend thread is started and joined on every `deno_execute`? I think it would be better if the backend thread was created and joined in `main()`.